test/test_nestedtensor.py

```diff
@@ -616,17 +616,32 @@ def test_nested_tensor_mul_in_place(self, device, dtype):
     def test_nested_tensor_sum_dim(self, device, dtype):
         params = ((2, (1, 1)), ((4), (4, 4)), (10, (3, 5, 7)))
 
-        def test_sum(nt, dim, keepdim=True):
+        def test_sum(device, dtype, ntensors, max_sizes, dim, keepdim=True, requires_grad=False):
+            nt = self.random_nt(device, dtype, ntensors, max_sizes)
             nt2 = nt.clone()
-            nt = nt.sum(dim=dim, keepdim=keepdim)
             ub2 = nt2.unbind()
-            ub2 = [t.sum(-1, keepdim=keepdim) for t in ub2]
-            nt2 = torch.nested_tensor(ub2)
-            self.assertEqual(nt, nt2)
+            if requires_grad:
+                nt.requires_grad_(requires_grad)
+                [t.requires_grad_(requires_grad) for t in ub2]
+            nt_sum = nt.sum(dim=dim, keepdim=keepdim)
+            ub2_sum = [t.sum(-1, keepdim=keepdim) for t in ub2]
+            self.assertEqual(nt_sum, torch.nested_tensor(ub2_sum))
+            # test backward
+            if requires_grad:
+                # generate gradient tensor that has the same size as the output
```

Comment:
It would be nice to have `torch.randn_like(nt_sum)` :D

Reply:
Could I hold off on this until Driss implements empty_like in #83140, I think that the implementation for randn_like might be similar

Reply:
Of course!

btw driss is using `nt = self.random_nt(device, dtype, 4, (4, 4))` in a few of his tests. Maybe that is a good replacement?

In any case, this is a small detail that can be done later.

Reply:
Ho you're already using that now :D